src/main/java/com/uber/profiling/Profiler.java

```diff
@@ -16,10 +16,20 @@
 
 package com.uber.profiling;
 
-public interface Profiler {
-    long getIntervalMillis();
+public abstract class Profiler {
+  private ProfilerRunner runner;
 
-    void setReporter(Reporter reporter);
+  public abstract long getIntervalMillis();
+  public abstract void setIntervalMillis(long millis);
 
-    void profile();
+  public abstract void setReporter(Reporter reporter);
+
+  public abstract void profile();
+
+  void setRunner(ProfilerRunner runner){
```

Comment:
Adding setRunner/getRunner here makes Profiler depends on ProfilerRunner, but it should not depend on it. Would suggest removing setRunner/getRunner here.

Reply:
Agree. Don't like that as well :)
Thinking of it - would it make sense to actually have the `ProfilerRunner` an inner class in the base `Profiler` abstract class or even better make the `Profiler` base class `Runnable` so it can be:
1. run once OR
2. scheduled
depending on the Profiler type

Again, my whole idea here is to make all profilers `runnable` (and hence cover both one-time and scheduled types).

Reply:
I see. I do not see your code call setRunner/getRunner though. Still prefer profiler not depending on this runner.